src/classes/Addresses.cls

```diff
@@ -0,0 +1,1161 @@
+
+    public void onAfterInsert() {
+        if (defaultAndSeasonalAddressesByHouseholdAccountId().size() > 0) {
+            refreshCurrentHHAddress(defaultAndSeasonalAddressesByHouseholdAccountId(), dmlWrapperLocal, false);
+        }
+
+        // we must process dmlWrapperLocal ourselves, so we can extract out the appropriate
+        // errors
+        // that might occur updating contacts and accounts, and associating those errors with
+        // the correct addresses.
+        ERR_Handler.Errors errors = TDTM_TriggerHandler.processDMLAllOrNothing
+                (dmlWrapperLocal, false);
+        if (errors.errorsExist) {
+            putErrorsOnAddresses(errors, defaultAndSeasonalAddressesByHouseholdAccountId(), new Map<Id, Address__c>());
+        }
+
+        // hack to detect empty addresses and delete them after we've propagated the change to all contacts and account
+        deleteEmptyAddresses();
+    }
+
+    public void onBeforeUpdate() {
+        updateLatestDatesForDefaultChanges();
+        verifyAddrHHAccOnly();
+        verifyAddrSeasonalNoOverlap();
+    }
+
+    public void onAfterUpdate() {
+        if (defaultAndSeasonalWithChanges().size() > 0) {
+            refreshCurrentHHAddress(defaultAndSeasonalWithChanges(), dmlWrapperLocal, false);
+        }
+
+        if (nonDefaultsWithChanges().size() > 0) {
+            updateAddrOverrides(nonDefaultsWithChanges(), dmlWrapperLocal);
+        }
+
+        // we must process dmlWrapper ourselves, so we can extract out the appropriate errors
+        // that might occur updating contacts and accounts, and associating those errors with
+        // the correct addresses.
+        ERR_Handler.Errors errors =
+                TDTM_TriggerHandler.processDMLAllOrNothing(
+                        dmlWrapperLocal, false);
+        if (errors.errorsExist) {
+            putErrorsOnAddresses(errors, defaultAndSeasonalWithChanges(), nonDefaultsWithChanges());
+        }
+    }
+
+    public void onAfterDelete() {
+        if (!addressesByHouseholdAccountId().isEmpty()) {
+            refreshCurrentHHAddress(addressesByHouseholdAccountId(), dmlWrapperLocal,
+                    false);
+        }
+
+        // we must process dmlWrapperLocal ourselves, so we can extract out the appropriate errors
+        // that might occur updating contacts and accounts, and associating those errors with
+        // the correct addresses.
+        ERR_Handler.Errors errors = TDTM_TriggerHandler.processDMLAllOrNothing(dmlWrapperLocal,
+                false);
+        if (errors.errorsExist) {
+            putErrorsOnAddresses(errors, addressesByHouseholdAccountId(), new Map<Id, Address__c>());
+        }
+    }
+
+    /********************************************************************************************************
+    * @description Retreive the defined default address type picklist value to use when creating a new
+    * Address record.
+    */
+    public static String defaultAddressType {
```

Comment:
There are a lot of Static methods in this class. As a domain class, shouldn't these be instance methods, or if they're just simple util methods they would go into a different class? Though, maybe that's your intent with this first iteration?

Reply:
Yes, most everything should be instance, I haven't changed many static vars/methods from static to instance or vice versa.  This is probably a result of moving this var. It came over from ADDR_Addresses_TDTM.defaultAddressType